HPCC-25753 Avoid unnecessary pod restarts on configMap changes

jakesmith opened the pull request:
Exclude sections from configSHA selectively, e.g. "global",
where known not to be used, to avoid unnecessary automatic pod
restarts.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

jakesmith committed:
HPCC-25753 Avoid unnecessary pod restarts on configMap changes

Exclude sections from configSHA selectively, e.g. "global",
where known not to be used, to avoid unnecessary automatic pod
restarts.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: bc553e80bcf06979e67b3b61281d135b82db5fbc
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-14.ca-central-1.compute.internal

jakesmith commented on common/workunit/workunit.cpp:
```diff
@@ -14240,17 +14240,17 @@ bool applyK8sYaml(const char *componentName, const char *wuid, const char *job,
         return false;
     }
-    jobYaml.replaceString("%jobname", jobname.str());
+    jobYaml.replaceString("_HPCC_JOBNAME_", jobname.str());
```
note to reviewer(s): was necessary to change our pseudo var substitution from leading '%' pattern, because as yaml, these cause invalid yaml issues.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-14.ca-central-1.compute.internal
Sha: bc553e80bcf06979e67b3b61281d135b82db5fbc
Build: success
Milestone:Install hpccsystems-platform-community_8.0.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |58 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |24 sec  (00:00:24) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |967 |967 |0 |897 sec  (00:14:57) |
|test (thor) |879 |879 |0 |1135 sec  (00:18:55) |
|test (roxie) |1038 |1038 |0 |905 sec  (00:15:05) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  600 sec (00:10:00) |  110 sec (00:01:50) |   20 sec (00:00:20) |   16 sec (00:00:16) | 1448 sec (00:24:08) |   18 sec (00:00:18) | 2220 sec (00:37:00) |

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: b3771e4698e6bb01931efee5351cc7f453d11b46
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-105.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-105.ca-central-1.compute.internal
Sha: b3771e4698e6bb01931efee5351cc7f453d11b46
Build: success
Milestone:Install hpccsystems-platform-community_8.0.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |59 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |967 |967 |0 |886 sec  (00:14:46) |
|test (thor) |879 |879 |0 |1120 sec  (00:18:40) |
|test (roxie) |1038 |1038 |0 |952 sec  (00:15:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  607 sec (00:10:07) |  113 sec (00:01:53) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1436 sec (00:23:56) |   19 sec (00:00:19) | 2221 sec (00:37:01) |

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: 114d0f713b7585a4faa0707b708b88a2f351f039
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-228.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-228.ca-central-1.compute.internal
Sha: 114d0f713b7585a4faa0707b708b88a2f351f039
Build: success
Milestone:Install hpccsystems-platform-community_8.0.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |59 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |24 sec  (00:00:24) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |967 |967 |0 |898 sec  (00:14:58) |
|test (thor) |879 |879 |0 |1139 sec  (00:18:59) |
|test (roxie) |1038 |1038 |0 |913 sec  (00:15:13) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  599 sec (00:09:59) |  107 sec (00:01:47) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1452 sec (00:24:12) |   18 sec (00:00:18) | 2222 sec (00:37:02) |

ghalliday commented on helm/hpcc/templates/_helpers.tpl:
```diff
@@ -1215,3 +1215,86 @@ Add a secret volume for a roxie udp key
+such that it will auto restart if the SHA changes.
+Uses filterConfig helper to select pertinent parts of the config to be part of the SHA.
+Pass in root, me, configMapHelper, component and excludeKeys
+excludeKeys is a comma separated list of key values to exclude from each secion, e.g. "global,esp.services,esp.queues"
```
secion->section

jakesmith replied:
Fixed

ghalliday reviewed (dismissed):
two trivial typos

jakesmith commented on helm/hpcc/templates/_helpers.tpl:
```diff
@@ -1215,3 +1215,86 @@ Add a secret volume for a roxie udp key
+
+Pass in root, me, configMapHelper, component, excludeSectionRegexList and excludeKeyList
+excludeSectionRegexList is a list of regexp's that filter out top-level sections, e.g. [".*spec.yaml$" ]
+excludeKeyList is a list of key values to exclude from each secion, e.g. [ "global", "esp.services" "esp.queues"]
```
Fixed

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Base branch changed.
Commit ID: 5ffb4be69b47287f1c2e899c6bf6708b252439a0
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-161.ca-central-1.compute.internal
 Git:  git version 2.9.5

jakesmith commented:
@ghalliday - rebased and fixed typo's. I have also added (last commit) an updated for storage groups (https://track.hpccsystems.com/browse/HPCC-25801), as I think should go in same time as this PR, as otherwise updates in storage groups that are dependent on initializeStorageGroups() would not be seen.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-161.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: 5ffb4be69b47287f1c2e899c6bf6708b252439a0
Build: success
Milestone:Install hpccsystems-platform-community_8.1.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |58 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |24 sec  (00:00:24) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |981 |981 |0 |889 sec  (00:14:49) |
|test (thor) |892 |892 |0 |1068 sec  (00:17:48) |
|test (roxie) |1050 |1050 |0 |1007 sec  (00:16:47) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  603 sec (00:10:03) |  106 sec (00:01:46) |   21 sec (00:00:21) |   19 sec (00:00:19) | 1380 sec (00:23:00) |   18 sec (00:00:18) | 2155 sec (00:35:55) |

ghalliday reviewed (dismissed):
Looks fine.  The atomics seem like a bit of overkill though - I'm not sure the function is ever called from more than one thread.  Could use a critical section instead.

jakesmith commented:
@ghalliday - squashed, please review

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: 74d8f43e820232b28bd78dc89d968ccb0969abf0
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-86.ca-central-1.compute.internal
 Git:  git version 2.9.5

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-86.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: 74d8f43e820232b28bd78dc89d968ccb0969abf0
Build: success
Milestone:Install hpccsystems-platform-community_8.1.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |59 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |0 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |6 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |24 sec  (00:00:24) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |981 |981 |0 |878 sec  (00:14:38) |
|test (thor) |892 |892 |0 |1040 sec  (00:17:20) |
|test (roxie) |1050 |1050 |0 |928 sec  (00:15:28) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  631 sec (00:10:31) |  111 sec (00:01:51) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1354 sec (00:22:34) |   19 sec (00:00:19) | 2161 sec (00:36:01) |

richardkchapman commented:
Looks like the definition of installConfigUpdateHook is wrong in win32 (in master):

unsigned installConfigUpdateHook(std::function<void ()> notifyFunc)

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: c840ca1a02e7c6cc8b19d86371f95120f236354b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-222.ca-central-1.compute.internal
 Git:  git version 2.9.5

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-222.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: c840ca1a02e7c6cc8b19d86371f95120f236354b
Build: success
Milestone:Install hpccsystems-platform-community_8.1.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |58 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |6 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |981 |981 |0 |868 sec  (00:14:28) |
|test (thor) |892 |892 |0 |1028 sec  (00:17:08) |
|test (roxie) |1050 |1050 |0 |922 sec  (00:15:22) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  636 sec (00:10:36) |  114 sec (00:01:54) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1342 sec (00:22:22) |   19 sec (00:00:19) | 2157 sec (00:35:57) |

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: f6566ed0efe7a8bce3a2c96f3341f8e59f3f2e55
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-212.ca-central-1.compute.internal
 Git:  git version 2.9.5

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-212.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: f6566ed0efe7a8bce3a2c96f3341f8e59f3f2e55
Build: success
Milestone:Install hpccsystems-platform-community_8.1.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |59 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |20 sec  (00:00:20) |
|test (hthor) |981 |981 |0 |909 sec  (00:15:09) |
|test (thor) |892 |892 |0 |1086 sec  (00:18:06) |
|test (roxie) |1050 |1050 |0 |966 sec  (00:16:06) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  632 sec (00:10:32) |  119 sec (00:01:59) |   22 sec (00:00:22) |   16 sec (00:00:16) | 1402 sec (00:23:22) |   19 sec (00:00:19) | 2219 sec (00:36:59) |

HPCCSmoketest commented:
Process of PR-14874, label: hpcc-25753-filter-configSHA starts now.
The reason of this test is: Forced to re-test.
Commit ID: 86f9b0c6f646a032d4fc726a3a25f227edda31a0
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-141.ca-central-1.compute.internal
 Git:  git version 2.9.5

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-141.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: 86f9b0c6f646a032d4fc726a3a25f227edda31a0
Build: success
Milestone:Install hpccsystems-platform-community_8.1.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |158 |158 |0 |0 |0 |59 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |20 sec  (00:00:20) |
|test (hthor) |981 |981 |0 |904 sec  (00:15:04) |
|test (thor) |892 |892 |0 |1079 sec  (00:17:59) |
|test (roxie) |1050 |1050 |0 |958 sec  (00:15:58) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  647 sec (00:10:47) |  111 sec (00:01:51) |   22 sec (00:00:22) |   16 sec (00:00:16) | 1395 sec (00:23:15) |   19 sec (00:00:19) | 2219 sec (00:36:59) |

jakesmith commented:
Now that https://github.com/hpcc-systems/HPCC-Platform/pull/14946 is merge, the win32 compile issue this PR hit is resolved.

@ghalliday - please check and merge.
